api-v2/README.md

```diff
@@ -0,0 +1,42 @@
+# GraduateNU API V2
+
+## Installation
+
+```bash
+npm install
+```
+
+## Running the app
+
+1. Run `yarn dev:db:up` from the project root to run the database in docker. You can run `yarn dev:db:down` from the project root to take down the database when you're done.
+
+2. Run `yarn dev` from within the api-v2 directory to run the server in watch mode.
+
+## Test
+
+```bash
+# unit tests
+$ yarn test
+
+# e2e tests
+$ yarn test:e2e
+
+# test coverage
+$ yarn test:cov
+```
+
+## Design Decisions
+
+A running log of design decisions that were taken. These can and will probably change as the server scales.
+
+- If you want to change the way things work, start a discussion, make the change and document it here!
+
+- If you're making changes to the codebase, please ensure that your changes follow the design documented here!
+
+### Error Handling
+
+- Services will be HTTP agnostic and not throw HTTP errors.
+
+- Services will not throw errors in general, and will return `null` for exceptional behavior. For instance, if a user with the same email already exists, return `null` and expect the controller to handle this.
```

Comment:
instead of "_will return `null` for exceptional behavior_" i think it should say "_will not intentionally throw exceptions_" – in the future we might want to distinguish certain errors by a different return type, for example:

```typescript
// in types.ts
type Result<T, E> = 
  | { type: "ok", ok: T }
  | { type: "err", err: E }

type LoginError = {
  msg: string,
  type: "baduser" | "badpass" | ...
}

// in code
let result: Result<User, LoginError> = ...
if (result.type === "ok") {
  return result.ok;
}
switch (result.err.type) {
  case "baduser": return new Error(`bad user: ${result.err.msg}`);
  case "badpass": return new Error(`bad pass .... `);
  ...
}
```

Reply:
A 100%. I'll change the text but I definitely anticipate the design section for error handling to change when we implement more specific error handling like you suggested! For now I just wanted to document the design our codebase currently follows with the intention that when we update the design we will change the doc.

Reply:
I think I'm actually going to leave it as is since I think the doc should specifically represent the design the codebase currently follows. I'll leave a note for how we should probably make the error handling better though!

Reply:
I think "_will not intentionally throw exceptions_" still captures the same meaning, because returning null is the same as not intentionally throwing an exception?

Reply:
True true, but wanted to be more specific as to exactly what we do now and why it isn't great. That essentially explicitly calls out this somewhat jank way of handling errors, which explicitly provides an opportunity for improvement and hence a change to the error design section 🤔